### PR TITLE
Add support for multiple trackers

### PR DESCRIPTION
**Pull request by MrBr**
Current implementation miss support for multiple trackers.

Changes: 
- Changed concept of GoogleAnalyticsTracker; now contains track ID within instance
- Moved static settings to separate class
- Change how code is organized because of all this changes
- Add support for tracker sampling rate
- Add backwards compatibility class to keep all previous versions running
- Create customDimensionsFieldsIndexMap to easy up customDimension implementation

I **haven't** changed methods signatures.

I've tested it locally but I would appreciate if others could also confirm that both, old and new implementations works. Tested on both Android and iOS.

I suggest that this default singleton export gets deprecated. It is more idiomatic that users create their own tracker instances.

Couldn't run example.

Let me know if I can do something more.

### CODE REVIEW

**Comment by cbrevik** [comment]
Hi @MrBr!

Wow, this is fantastic work. I've been thinking about supporting multiple trackers for some time now. I'm happy to see this PR!

I'm busy all day tomorrow, but I'll take some time this evening and this weekend to review and test this. Again, great work, thanks!

File: src/GoogleAnalyticsBackwardsCompability.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  }
+
+  trackPurchaseEvent(product = {}, transaction = {}, eventCategory = "Ecommerce", eventAction = "Purchase") {
+    this.tracker.trackPurchaseEvent(...arguments);
+  }
+
+  trackMultiProductsPurchaseEvent(products = [], transaction = {}, eventCategory = "Ecommerce", eventAction = "Purchase") {
+    this.tracker.trackMultiProductsPurchaseEvent(...arguments);
+  }
+
```
Comment (cbrevik) [inline src/GoogleAnalyticsBackwardsCompability.js]: A new method called [trackMultiProductsPurchaseEventWithCustomDimensionValues](https://github.com/idehub/react-native-google-analytics-bridge/blob/master/index.js#L94) is lacking from the implementation here.

File: src/GoogleAnalyticsTracker.js [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+   * Track a purchase event. This uses the Enhanced Ecommerce GA feature.
+   * @param  {Array} products       An array with products
+   * @param  {Object} transaction   An object with transaction values
+   * @param  {String} eventCategory The event category, defaults to Ecommerce
+   * @param  {String} eventAction   The event action, defaults to Purchase
+   */
+  trackMultiProductsPurchaseEvent(products = [], transaction = {}, eventCategory = "Ecommerce", eventAction = "Purchase") {
+    GoogleAnalyticsBridge.trackMultiProductsPurchaseEvent(this.id, ...arguments);
+  }
+
```
Comment (cbrevik) [inline src/GoogleAnalyticsTracker.js]: [trackMultiProductsPurchaseEventWithCustomDimensionValues](https://github.com/idehub/react-native-google-analytics-bridge/blob/master/index.js#L94) also lacking from the implementation here.

Comment (MrBr) [inline src/GoogleAnalyticsTracker.js]: Added as well.

**Comment by MrBr** [comment]
It is no problem, I have updated PR.

**Comment by cbrevik** [comment]
Great, thanks a lot! I will see about releasing this in a major version tomorrow or friday.

**Pull request merged**